### PR TITLE
Added authorization and reorganized solution.

### PR DESCRIPTION
**Pull request by cestewart**
Added authorization and reorganized solution.

### CODE REVIEW

File: AutoRenter.Api/Authorization/IsAdminHandler.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+
+namespace AutoRenter.Api.Authorization
+{
+    public class IsAdminHandler : AuthorizationHandler<IsAdminRequirement>
```
Comment (BWortman) [inline AutoRenter.Api/Authorization/IsAdminHandler.cs]: Could this be renamed to 'AdminHandler'? IsAdminHandler sounds more like a method name than a class name, doesn't it? But maybe it's a standard for this sort of thing...?

Comment (cestewart) [inline AutoRenter.Api/Authorization/IsAdminHandler.cs]: IsAdmin is not a great name.  It breaks my own rule and really should be IsAdministrator.

I don't think I could look at names and tell you which names are for classes and which ones are for methods.

The attribute I created is [Authorize(Policy="IsAdmin")].  This lines up nicely with IsAdminRequirement and IsAdminHandler.  

How about I change it to  [Authorize(Policy="IsAdministrator")].  And rename the classes to IsAdministratorAuthorizationHandler and IsAdministratorAuthorizationRequirement?

Comment (BWortman) [inline AutoRenter.Api/Authorization/IsAdminHandler.cs]: I like that, yes (remembering your recommendation on avoiding abbreviations), but also how about simply dropping the "Is" prefix from both of the class names?

E.g.
`options.AddPolicy("IsAdministrator", policy => policy.Requirements.Add(new AdministratorAuthorizationRequirement()));`

File: AutoRenter.Api/Features/Location/LocationsController.cs [hunk 2]
```diff
@@ -22,6 +23,7 @@ public LocationsController(IResponseConverter responseConverter, IMediator media
         }
 
         [HttpGet]
+        [AllowAnonymous]
```
Comment (BWortman) [inline AutoRenter.Api/Features/Location/LocationsController.cs]: Probably a dumb question, BUT, these AllowAnonymous attributes are here only temporarily, right?

File: AutoRenter.Api/AppSettings.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿namespace AutoRenter.Api
+{
+    public class AppSettings
+    {
+        public virtual bool InMemoryProvider { get; set; }
+
+        public virtual TokenSettings TokenSettings { get; set; }
+    }
+
+    public class TokenSettings
```
Comment (BWortman) [inline AutoRenter.Api/AppSettings.cs]: Prefer to move this to a separate file.

File: AutoRenter.Api/Authorization/IsAdminHandler.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, IsAdminRequirement requirement)
+        {
+            if (!context.User.HasClaim(c => c.Type == "is_administrator"))
```
Comment (BWortman) [inline AutoRenter.Api/Authorization/IsAdminHandler.cs]: How about introducing a constant for the "is_administrator" string...

File: AutoRenter.Api/Features/Location/LocationsController.cs (hunk 2 above)
Comment (cestewart) [inline AutoRenter.Api/Features/Location/LocationsController.cs]: Yes.  This is just so everything keeps working.

File: AutoRenter.Api/AppSettings.cs (hunk 3 above)
Comment (cestewart) [inline AutoRenter.Api/AppSettings.cs]: Done

File: AutoRenter.Api/Authorization/IsAdminHandler.cs (hunk 4 above)
Comment (cestewart) [inline AutoRenter.Api/Authorization/IsAdminHandler.cs]: I created a struct AutoRenterClaimNames with constants for all the claim names in it.

**Pull request merged**